docs/contracts.rst

```diff
@@ -1005,10 +1005,15 @@ virtual method lookup.
 
 .. index:: ! constructor
 
+.. _constructor:
+
 Constructors
 ============
 
-A constructor is an optional function declared with the ``constructor`` keyword which is executed upon contract creation.
+A constructor is an optional function declared with the ``constructor`` keyword which is executed upon contract creation, and designed to run contract initialisation code. It is always generates code that initialises any state variables you declare to their specified default value if you initialise them inline, or zero if you do not.
+
+After that it returns the final code of the contract. The final deployment of the code costs additional gas linear to the length of the code, if you did not supply enough gas to initiate the state variables declared in the constructor, then an "out of gas" exception is generated. This is the **only** situation where an "out of gas" exception does not revert changes to the state.
```

Comment:
Why do you mention "enough gas to initiate the state variables declared in the constructor"?

Also I'm pretty sure state changes are indeed reverted.

Reply:
@chriseth Most of this relates back to the original change from the FAQ page, first it stated that:

> This is the **only** situation where an "out of gas" exception does not revert changes to the state.

So maybe that's not true anymore (or never was). If it's not true then the other statement:

> if you did not supply enough gas to initiate the state variables declared in the constructor, then an "out of gas" exception is generated.

Is kind of irrelevant, but it was based on our discussion where it was said that an out of gas exception is generated if you don't supply enough gas to initiate all the state variables. Maybe it would be more accurate to say that it's one of the things that consumes gas, as it's unlikely someone will create a contract that has nothing but state variables?